src/framework/actions/turn.ts

```diff
@@ -1,13 +1,120 @@
-import { GameState } from "../types";
+import { CardID, GameState } from "../types";
+import { discardAll, drawCard } from "./hand";
+import { MAX_GOLD } from "../constants";
+
+const handleReadyPhase = ($: GameState): void => {
+  if ($.turnPhase === "READY") {
+    const P = $.players[$.activePlayer];
+
+    P.hasShuffledThisTurn = false;
+    P.hasBuiltWorkerThisTurn = false;
+
+    P.patrol.squadLeader = null;
+    P.patrol.elite = null;
+    P.patrol.scavenger = null;
+    P.patrol.technician = null;
+    P.patrol.lookout = null;
+
+    for (const I of Object.values($.instances)) {
+      if (I.owner === $.activePlayer) {
+        I.arrivalFatigue = false;
+
+        I.readyState = I.readyState === "DISABLED" ? "EXHAUSTED" : "READY";
+
+        // Make dead heroes available
+      }
+    }
+
+    $.turnPhase = "UPKEEP";
+
+    handleUpkeepPhase($);
+  } else {
+    throw new Error(`not in ready phase (${$.turnPhase})`);
```

Comment:
nit: Here and at other places where we have a condition to do nothing but throw, it seems like we'd rather do
```
if (!valid) {
  throw new Error(msg);
}

// ... impl
```

Avoiding the if-else let's us not worry about the negative condition ordering, and save an indent level for easier visual parsing

Reply:
I avoid lone if statement guards like that; I generally prefer to have things nested with the error cases at the end. But since that goes against the convention in this project, I'll adjust things here.